doc/site/README.md

```diff
@@ -1,31 +1,143 @@
-# Recoil Site
+# Setting Up Local Site Development
 
-The recoil site is a jekyll application, usually hosted on github pages.
+This version of the site was designed to be as simple to edit as possible. In the majority of cases, all you need is HUGO itself.
 
-## Testing locally
+## Writing articles for the site
 
-Install Ruby 3.2.3 ([newer versions can cause problems](https://stackoverflow.com/a/77896791/317135)).
+All you need is `hugo-extended`, available now at your local package manager.
 
-Inside `doc/site` run:
+`cd` into `doc/site` and run `hugo server`. It will take 2-5 seconds to generate the site, but then any changes you make will be compiled quickly. The site will be hosted on [Localhost](http://localhost:1313/RecoilEngine).
 
+It's worth noting that since it's an incremental compile by default, if you start getting odd behavior, restart the server to build it fresh. This happens a lot if you're messing with partials or layouts.
+
+## Editing Lua Docs
+
+if you are editing the Lua docs, the process can be a little longwinded. They need to be extracted from the C++ files, compiled into markdown, and then moved.
+
+Doing this locally requires:
+1. Rust, for cargo
+2. Node, for NPM (or equivalent software repo)
+3. Python 3, latest version, for glue code
+
+And the following programs:
+1. Hugo extended, latest version, installed from your favorite package manager
+2. Emmylua_doc_cli, 0.7.2 or greater, installed with cargo
+3. lua-doc-extractor by resident rhys_vdw
+
+Install [HUGO](https://gohugo.io/installation/)
```

Comment:
```suggestion
1. [Hugo extended](https://gohugo.io/installation/)
2. [emmylua_doc_cli](https://github.com/CppCXY/emmylua-analyzer-rust/tree/main/crates/emmylua_doc_cli)
3. [lua-doc-extractor](https://github.com/rhys-vdw/lua-doc-extractor)
```
 
Probably not worth specifying version numbers. Also you don't know that Hug should be latest version, what if we don't upgrade for five years and want an older version. IMO much cleaner to just link to the readmes that show up-do-date installation instructions for whichever platform.

Feel free to reject if you disagree.